ksql-engine/src/main/java/io/confluent/ksql/materialization/KsqlMaterializationFactory.java

```diff
@@ -144,7 +162,19 @@ public Materialization create(
     );
   }
 
-  private static ValueMapperFactory defaultValueMapperFactory() {
+  private static AggregateMapperFactory defaultAggregateMapperFactory() {
+    return (info, functionRegistry) ->
+        new AggregateParams(
+            info.schema(),
+            info.startingColumnIndex(),
+            functionRegistry,
+            info.aggregateFunctions()
+        )
+            .getAggregator()
+            .getResultMapper()::apply;
```

Comment:
nit: i'd refactor `getResultMapper` to return a `Function` and build the `ValueMapper` we pass to streams from that.

Reply:
Agreed, but not in this PR.